From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Kleinschmidt <kleinschmidtorama@gmail.com>
Date: Mon, 23 Sep 2019 17:07:53 -0400
Subject: add zoom limit setters to webcontents

Allows minimum_zoom_percent_ and maximum_zoom_percent_ to be set on WebContents.
This is needed by Electron to allow apps to limit how much an app can zoom in
or out.

diff --git a/content/browser/web_contents/web_contents_impl.cc b/content/browser/web_contents/web_contents_impl.cc
index 6bbb07698eb21f03c48d88aa1c67e138302bf786..d831d44295a2c0341279c34e8368eeffa64e2783 100644
--- a/content/browser/web_contents/web_contents_impl.cc
+++ b/content/browser/web_contents/web_contents_impl.cc
@@ -4173,10 +4173,18 @@ bool WebContentsImpl::GetClosedByUserGesture() {
   return closed_by_user_gesture_;
 }
 
+void WebContentsImpl::SetMinimumZoomPercent(int zoom_percent) {
+  minimum_zoom_percent_ = zoom_percent;
+}
+
 int WebContentsImpl::GetMinimumZoomPercent() {
   return minimum_zoom_percent_;
 }
 
+void WebContentsImpl::SetMaximumZoomPercent(int zoom_percent) {
+  maximum_zoom_percent_ = zoom_percent;
+}
+
 int WebContentsImpl::GetMaximumZoomPercent() {
   return maximum_zoom_percent_;
 }
diff --git a/content/browser/web_contents/web_contents_impl.h b/content/browser/web_contents/web_contents_impl.h
index 970ffc6a05bdf5a5e1c78dab5d6e35bd009014a5..7b4082ac776fc78b4e3d220b0850140ed055635a 100644
--- a/content/browser/web_contents/web_contents_impl.h
+++ b/content/browser/web_contents/web_contents_impl.h
@@ -457,7 +457,9 @@ class CONTENT_EXPORT WebContentsImpl : public WebContents,
   void SystemDragEnded(RenderWidgetHost* source_rwh) override;
   void SetClosedByUserGesture(bool value) override;
   bool GetClosedByUserGesture() override;
+  void SetMinimumZoomPercent(int zoom_percent) override;
   int GetMinimumZoomPercent() override;
+  void SetMaximumZoomPercent(int zoom_percent) override;
   int GetMaximumZoomPercent() override;
   void SetPageScale(float page_scale_factor) override;
   gfx::Size GetPreferredSize() override;
@@ -1730,8 +1732,8 @@ class CONTENT_EXPORT WebContentsImpl : public WebContents,
   bool closed_by_user_gesture_;
 
   // Minimum/maximum zoom percent.
-  const int minimum_zoom_percent_;
-  const int maximum_zoom_percent_;
+  int minimum_zoom_percent_;
+  int maximum_zoom_percent_;
 
   // Used to correctly handle integer zooming through a smooth scroll device.
   float zoom_scroll_remainder_;
diff --git a/content/public/browser/web_contents.h b/content/public/browser/web_contents.h
index dcc3604207b472f7ffa65a6768e2f87cca4c9952..8120828ea54b7ba1c283bf7a569b8a556a88d94c 100644
--- a/content/public/browser/web_contents.h
+++ b/content/public/browser/web_contents.h
@@ -841,8 +841,10 @@ class WebContents : public PageNavigator,
   virtual void SetClosedByUserGesture(bool value) = 0;
   virtual bool GetClosedByUserGesture() = 0;
 
-  // Gets the minimum/maximum zoom percent.
+  // Gets/sets the minimum/maximum zoom percent.
+  virtual void SetMinimumZoomPercent(int zoom_percent) = 0;
   virtual int GetMinimumZoomPercent() = 0;
+  virtual void SetMaximumZoomPercent(int zoom_percent) = 0;
   virtual int GetMaximumZoomPercent() = 0;
 
   // Set the renderer's page scale to the given factor.
